runtime-modules/working-group/src/tests/mod.rs

```diff
@@ -1165,6 +1166,131 @@ fn update_worker_role_account_fails_with_invalid_origin() {
+
+        EventFixture::assert_last_crate_event(RawEvent::WorkerStorageUpdated(
+            worker_id,
+            storage_field,
+        ));
+    });
+}
+
+#[test]
+fn update_worker_storage_by_leader_succeeds() {
+    build_test_externalities().execute_with(|| {
+        let storage_field = vec![0u8].repeat(10);
+
+        let worker_id = HireLeadFixture::default().hire_lead();
+
+        let old_lead = TestWorkingGroup::worker_by_id(worker_id);
+
+        let update_storage_fixture = UpdateWorkerStorageFixture::default_with_storage_field(
+            worker_id,
+            storage_field.clone(),
+        );
+
+        update_storage_fixture.call_and_assert(Ok(()));
+
+        let new_lead = TestWorkingGroup::worker_by_id(worker_id);
+
+        assert_eq!(
+            new_lead,
+            Worker {
+                storage: storage_field,
+                ..old_lead
+            }
+        );
+    });
+}
+
+#[test]
+fn update_worker_storage_fails_with_invalid_origin_signed_account() {
+    build_test_externalities().execute_with(|| {
+        let worker_id = fill_default_worker_position();
+
+        let storage_field = vec![0u8].repeat(10);
+
+        let update_storage_fixture =
+            UpdateWorkerStorageFixture::default_with_storage_field(worker_id, storage_field)
+                .with_origin(RawOrigin::Signed(2));
+
+        update_storage_fixture.call_and_assert(Err(
+            Error::<Test, TestWorkingGroupInstance>::SignerIsNotWorkerRoleAccount.into(),
+        ));
+    });
+}
+
+#[test]
+fn update_worker_storage_fails_with_invalid_worker_id() {
+    build_test_externalities().execute_with(|| {
+        let storage_field = vec![0u8].repeat(10);
+
+        fill_default_worker_position();
+
+        let invalid_worker_id = 1;
+
+        let update_storage_fixture = UpdateWorkerStorageFixture::default_with_storage_field(
+            invalid_worker_id,
+            storage_field.clone(),
+        );
+
+        update_storage_fixture.call_and_assert(Err(
+            Error::<Test, TestWorkingGroupInstance>::WorkerDoesNotExist.into(),
+        ));
+    });
+}
+
+#[test]
+fn update_worker_storage_fails_with_too_long_text() {
+    build_test_externalities().execute_with(|| {
+        let storage_field = vec![0u8].repeat(default_text_constraint().max() as usize + 1);
+
+        let worker_id = fill_default_worker_position();
+
+        let update_storage_fixture = UpdateWorkerStorageFixture::default_with_storage_field(
+            worker_id,
+            storage_field.clone(),
+        );
+
+        update_storage_fixture
+            .call_and_assert(Err(DispatchError::Other("WorkerStorageTextTooLong")));
+    });
+}
+
+#[test]
+fn update_worker_storage_fails_with_too_short_text() {
+    build_test_externalities().execute_with(|| {
+        let worker_id = fill_default_worker_position();
+
+        let storage_field = vec![0u8].repeat(default_text_constraint().min as usize - 1);
```

Comment:
If we want to always allow 0 size it wouldn't make sense to have this test.

Reply:
Fixed it here https://github.com/Joystream/joystream/pull/2180/commits/1340c1f031aadbcdf0ca95a5409bdaff25ae6f04